Feature copter parasite wing

Eddur99 opened the pull request:
This function calculates the aerodynamic drag characteristics of a wing through a series of empirical and theoretical calculations, including factors for skin friction,
compressibility, and interference. The function uses inputs such as wing span, chord length, wing thickness, and aspect ratio from a configuration dictionary. 
It also takes into account freestream conditions like Mach number and Reynolds number.It computes the dimensional and non dimensional drag coefficient.

Eddur99 committed:
Copter_parasite_wing

Eddur99 committed:
Create Pioneer300.py

Eddur99 committed:
Add files via upload

EttoreSaetta commented on Copter_parasite_wing.py:
```diff
@@ -0,0 +1,49 @@
+"""
+This function calculates the aerodynamic drag characteristics of a wing through a series of empirical and theoretical calculations, including factors for skin friction,
+compressibility, and interference. The function uses inputs such as wing span, chord length, wing thickness, and aspect ratio from a configuration dictionary. 
+It also takes into account freestream conditions like Mach number and Reynolds number.It computes the dimensional and non dimensional drag coefficient.
+
+The theory behind this code can be found in: Carlo De Nicola - "Appunti per il corso di Aerodinamica degli Aeromiboili" - a.a. 2018/2019 - Appendice E.
+
+    Authors: Eduardo Duraccio, Samuel Filosa
+    Date: 17/07/2024
+    Version: 1.01
+
+"""
```
This comment is with `#`

EttoreSaetta commented on TestCases/Copter_parasite_wing/Black_Hawk.py:
```diff
@@ -0,0 +1,102 @@
+"""
+This test case script calculates aerodynamic coefficients for the horizontal stabilizer of a simulated Sirosky UH-60 Black Hawk using Python.
+The script utilizes parameters such as wing span, wing chord, wing thickness ratio, and aspect ratio specific to the copter selected, all defined in a configuration dictionary (config).
+Additionally, flight parameters such as cruise speed, altitude, speed of sound, kinematic viscosity, and density are specified to compute the Mach number and Reynolds number necessary for aerodynamic calculations.
+
+Author: Eduardo Duraccio
+    Date: 17/07/2024
+    Version: 1.02
+"""
```
This comment is with `#`

EttoreSaetta commented on TestCases/Copter_parasite_wing/Pioneer300.py:
```diff
@@ -0,0 +1,102 @@
+"""This test case script calculates aerodynamic coefficients for the wing of a simulated Pioneer 300 aircraft using Python.
+The script utilizes parameters such as wing span, wing chord, wing thickness ratio, and aspect ratio specific to the Pioneer 300, all defined in a configuration dictionary (config).
+ Additionally, flight parameters such as cruise speed, altitude, speed of sound, kinematic viscosity, and density are specified to compute the Mach number and Reynolds number necessary for aerodynamic calculations.
+
+ Author: Eduardo Duraccio
+    Date: 17/07/2024
+    Version: 1.02
+"""
```
This comment is with `#`

EttoreSaetta reviewed (changes requested):
- I would create a new class, or update the `Helicopter.py` class with your additional input.
- Do not use the "dictionary" as input, but the class.
- In the test cases do not re-write the function, but import your function and use it.

Eddur99 committed:
Update Copter_parasite_wing.py

Commented using #

Eddur99 committed:
Update Black_Hawk.py

Commented using #

Eddur99 committed:
Update Copter_parasite_wing.py

Modified Function copter_parasite_wing:
#This function calculates the aerodynamic drag characteristics of a wing through a series of empirical and theoretical calculations, including factors for skin friction,
#compressibility, and interference. The function uses inputs such as wing span, chord length, wing thickness, and aspect ratio from a configuration dictionary. 
#It also takes into account freestream conditions like Mach number and Reynolds number.It computes the dimensional and non dimensional drag coefficient.

Eddur99 committed:
Update Black_Hawk.py

Test case Balck Hawk UH-60

Eddur99 committed:
Update Pioneer300.py

Test case Pioneer 300

Eddur99 committed:
Add files via upload

Eddur99 committed:
Restore Helicopter_Class.py

Eddur99 committed:
Add Helicopter horizontal stabilizer class

Eddur99 committed:
Update Black_Hawk.py

Eddur99 committed:
Update Pioneer300.py

EttoreSaetta commented on Helicopter_HS_Class.py:
```diff
@@ -0,0 +1,35 @@
+        self.R_mr=R_mr
+        self.M_inf = None
+        self.Re_inf = None
+        self.ala_fissa = ala_fissa
+        self.profilo = profilo
```
Switch to English

EttoreSaetta commented on Copter_parasite_wing.py:
```diff
@@ -0,0 +1,81 @@
+    S_rotor= 3.14*r_mr**2 # reference area (rotor area)
+    Swet = 2 * S  # wetted area of the wing (both sides)
+
+    if profilo == "4-series":                       # insert "4-series" to use 4-series airfoil
+        FF = 1 + 1.68 * (tau / c) + 3 * (tau / c)**2    # form factor for 4-series airfoil
+    elif profilo == "biconvex":                       # insert "biconvex" to use biconvex airfoil
+        FF= 1 + 1.20 * tau + 100 * tau**4               # form factor for biconvex-airfoil
+    elif profilo == "6-series":                      # insert "6-series" to use 6-series airfoil
+        FF = 1 + 1.44 * tau + 2 * tau**2                # form factor for 6-series airfoil
```
Switch to English

Eddur99 committed:
Update Copter_parasite_wing.py

Eddur99 committed:
Update Helicopter_HS_Class.py

Eddur99 committed:
Update Black_Hawk.py

Eddur99 committed:
Update Pioneer300.py
